Source/ResponseSerialization.swift

```diff
@@ -145,11 +145,13 @@ extension Request {
         - returns: A data response serializer.
     */
     public static func dataResponseSerializer() -> ResponseSerializer<NSData, NSError> {
-        return ResponseSerializer { _, _, data, error in
+        return ResponseSerializer { _, response, data, error in
             guard error == nil else { return .Failure(error!) }
 
-            guard let validData = data where validData.length > 0 else {
-                let failureReason = "Data could not be serialized. Input data was nil or zero length."
+            guard response?.statusCode ?? -1 != 204 else { return .Success(NSData()) }
```

Comment:
Yeah now I remember why I didn't do this...if response is nil then the `guard` will fail and it will return a `.Success` case which is definitely not what we want. I don't see an cleaner way to do it than I already have. Open to suggestions...